### ISSUE TITLE
App randomly crashes in production

### ISSUE THREAD

**Issue by Wetsie** [avishayil/react-native-restart#161]
It almost everytime crashes without any logs, but once I've managed to catch them:

```
E ReactNativeJS: TypeError: undefined is not a function, js engine: hermes
I ReactNativeJS: Unable to symbolicate stack trace: Bundle was not loaded from Metro.
E ReactNativeJS: Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
E ReactNativeJS: This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native., js engine: hermes
I ReactNativeJS: Unable to symbolicate stack trace: Bundle was not loaded from Metro.
E ReactNativeJS: Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication). A frequent cause of the error is that the application entry file path is incorrect.
E ReactNativeJS: This can also happen when the JS bundle is corrupt or there is an early initialization error when loading React Native., js engine: hermes
I ReactNativeJS: Unable to symbolicate stack trace: Bundle was not loaded from Metro.
```

It looks like the problem appeared after updating react native to version 0.64.0. Does this lib support new version or hermes engine?

**Comment by avishayil** [comment]
I'm not familiar with this issue, it happens only on production bundles?

**Comment by vikasverma92** [comment]
https://github.com/software-mansion/react-native-screens/issues/17#issuecomment-424704067

this worked for me

**Comment by avishayil** [comment]
I'm checking compatibility issues with hermes

**Comment by saskaak** [comment]
Any news on this? Crashes on production build with React Native 0.63 / Android / Hermes.

**Comment by eddievlagea** [comment]
@saskaak does it still crash on production build? I'm considering using this library but these crashes make me nervous.

**Comment by saskaak** [comment]
> @saskaak does it still crash on production build? I'm considering using this library but these crashes make me nervous.

For us, yes. Works just fine on iOS.

**Comment by francois-pasquier** [comment]
Works ok on iOS, sadly still broken on Android...

Did anyone find a good alternative to this library?

**Comment by francois-pasquier** [comment]
I don't know if the issue is the same but I had crashes everytime on Android (mostly because of react-navigate-geture-handler).

A fix is instead of restarting as soon as the user press a button, just add some logic to unmount all your components.

RootNavigation.tsx:
```tsx
const RootNavigation = () => {
	const { shouldRestart } = useShouldRestart()

    if (shouldRestart) {
        return <Restart />
    }
    
    // return your navigators
}
```

Restart.tsx:
```tsx
import { useMount } from 'ahooks'
import { View } from 'react-native'
import CodePush from 'react-native-code-push'

const Restart = () => {
    useMount(() => {
        // restartApp()
    })

    return <View />
}

export default Restart
```

useShouldRestart.ts:
```ts
import { atom, useRecoilState } from 'recoil'

export const shouldRestartAtom = atom({
    key: 'shouldRestart',
    default: false
})

const useShouldRestart = () => {
    const [shouldRestart, setShouldRestart] = useRecoilState(shouldRestartAtom)

    return { shouldRestart, setShouldRestart }
}

export default useShouldRestart
```

With this, you can simply set `shouldRestart` to true and the app should cleanly close itself.
I still have `SoftExceptions`s popping on logcat but the app doesn't crash anymore.

**Comment by saad-saadi** [comment]
I solved the issue by https://github.com/avishayil/react-native-restart/issues/162#issuecomment-1038724027

**Comment by PrinceUBS** [comment]
> I solved the issue by [#162 (comment)](https://github.com/avishayil/react-native-restart/issues/162#issuecomment-1038724027)

Facing same issue in production build... I tried above solution but it's not working perfectly(shared video)

https://user-images.githubusercontent.com/75003713/160824634-26970880-f480-4319-b0d2-516b1fdacf47.mp4

- app freezes for little-bit time & throw error,
- there is not others things in my function 
### code:
```
    const setLanguage = async (code: string) => {
        if (code === 'ar') {
            await i18next.changeLanguage(code).then(() => {
                I18nManager.forceRTL(true);
                setTimeout(() => { RNRestart.Restart(); }, 150);
                setShowSelectionPopup(false);
            });
        } else {
            await i18next.changeLanguage(code).then(() => {
                I18nManager.forceRTL(false);
                setTimeout(() => { RNRestart.Restart(); }, 150);
                setShowSelectionPopup(false);
            });
        }
        console.log("code:", code);
    };

```
@avishayil

**Comment by thanveershah** [comment]
> > I solved the issue by [#162 (comment)](https://github.com/avishayil/react-native-restart/issues/162#issuecomment-1038724027)
> 
> Facing same issue in production build... I tried above solution but it's not working perfectly(shared video)
> 
>  RN-Restart-rec.mp4 
> * app freezes for little-bit time & throw error,
> * there is not others things in my function
> 
> ### code:
> ```
>     const setLanguage = async (code: string) => {
>         if (code === 'ar') {
>             await i18next.changeLanguage(code).then(() => {
>                 I18nManager.forceRTL(true);
>                 setTimeout(() => { RNRestart.Restart(); }, 150);
>                 setShowSelectionPopup(false);
>             });
>         } else {
>             await i18next.changeLanguage(code).then(() => {
>                 I18nManager.forceRTL(false);
>                 setTimeout(() => { RNRestart.Restart(); }, 150);
>                 setShowSelectionPopup(false);
>             });
>         }
>         console.log("code:", code);
>     };
> ```
> 
> @avishayil

Where you able to fix this issue?

**Comment by PrinceUBS** [comment]
> Where you able to fix this issue?


### Yep! by changing the Build configuration...

_Open_ **xCode > Product > Scheme > Edit Scheme... > Run(select from left panel) > Build configuration(info tab) = Release** _(this xCode 12.4 attached SS)_
![image_2022_05_10T04_31_42_681Z](https://user-images.githubusercontent.com/75003713/167544311-5ab21ca8-809f-4dd5-9096-480cc4a61eff.png)

**Comment by HammzaHM** [comment]
I've made a PR for it #211, I've used [ProcessPhoenix](https://github.com/JakeWharton/ProcessPhoenix) framework instead of the `react-native-code-push` code style.

**Comment by avishayil** [comment]
Merged #211
